mdpow/analysis/ensemble.py

```diff
@@ -469,14 +469,14 @@ def _single_universe(self):
             Run on each universe in the ensemble during when
             self.run in called.
         """
-        pass  # pragma: no cover
+        raise NotImplementedError
 
     def _single_frame(self):
         """Calculate data from a single frame of trajectory
 
         Called on each frame for universes in the Ensemble.
         """
-        pass  # pragma: no cover
+        raise NotImplementedError
```

Comment:
Add test that checks that this NotImplementedError is raised. If necessary by calling the method explicitly
```python
def test_single_frame_raises_NotImplementedError():
    ...
    ea = EnsembleAnalysis(...)
    with pytest.raises(NotImplementedError):
          ea._single_frame()
```

Reply:
https://github.com/Becksteinlab/MDPOW/blob/ensemble_run_update/mdpow/tests/test_dihedral.py
- file does not exist in pull request because it was deleted and moved to another pull request, so I just linked the source from the branch
- function is at end of module
- this is the explicit call of _single_universe() to test NotImplementedError is raised when not defined, but specifically moved for us in DihedralAnalysis as instructed

I am going to add explicit tests for raising NotImplementedError for _single_frame() and _single_universe() similar to pattern in your comment above, placed in test_ensemble.py, because test_run.py is for the partition coefficient calculation and test_ensemble.py deals with EnsembleAnalysis class where this run method that calls single_frame/uni is defined
- will be in this pull request

Reply:
Look into https://github.com/Becksteinlab/MDPOW/pull/216/files and you see the annotation by codecov that says that the line was not covered by tests.

Also check the codecov report in the checks and ultimately https://app.codecov.io/gh/Becksteinlab/MDPOW/pull/216